doc/design/status_port.md

```diff
@@ -0,0 +1,453 @@
+# StatusPort
+
+## Summary and requirements
+
+The `StatusPort` is an alternative to the publish subscriber communication in `iceoryx_posh`.
+
+The target of the `StatusPort` are use-case with the following properties:
+
+* Data is transferred once or is rarely updated
+* There are many readers which are interested in the data
+* Data can be persistent
+    * The lifetime of the transferred data is bound to the lifetime of `StatusPortData`
+* Readers don't access the data directly but via a lambda
+    * Preventing torn reads, since the `StatusPortReader` detects if the data
+    changed during `take()` operation (Frankenstein check) and re-executes the lambda
+* Has to be attachable to `Listener`
+* `StatusPort` could potentially be used to communicate between different ASIL domains
+
+Potential applications are:
+
+* Introspection topics
+* Service discovery
+
+## Design
+
+### Terminology
+
+* Transaction: Atomic state of the world, which is changed by a write operation
+* Current transaction: Transaction in local scope, read in the beginning of each operation
+* Latest transaction: Transaction in the shared memory managment segment
+* Chunk: Untyped piece of memory located in the shared memory payload segment
+* Read position: The chunk, which was used the last time to write data
+* Write position: The opposite chunk, not currently being used by the `StatusPortReader`s
+
+### Contract & properties
+
+* 1:n communication aka broadcast
+* Two memory chunks from shared memory payload segment are used
+* Data exchanged needs to be trivially-copyable
+* Storing a chunk cannot fail
```

Comment:
Maybe add here why this is. Since the underlying memory is trivially copyable and transferred via `memcpy`

Reply:
Not sure if I got you here. `memcpy` is not used, instead the de-referenced pointer is passed to the `callable`. I added the "why".